treescript/src/treescript/gecko/android_l10n.py

```diff
@@ -0,0 +1,206 @@
+"""Treescript android-l10n import and sync support.
+"""
+import asyncio
+import logging
+import os
+import shutil
+import subprocess
+import tempfile
+from compare_locales import parser, paths
+
+from scriptworker_client.aio import retry_async
+
+from treescript.gecko import mercurial as vcs
+from treescript.util.task import CLOSED_TREE_MSG, DONTBUILD_MSG, get_dontbuild, get_ignore_closed_tree, get_android_l10n_import_info, get_android_l10n_sync_info
+from treescript.util.treestatus import check_treestatus
+
+log = logging.getLogger(__name__)
+
+
+# get_android_l10n_files_toml {{{1
+def get_android_l10n_files_toml(toml_path, search_path=None):
+    """Extract list of localized files from project configuration (TOML)"""
+
+    basedir = os.path.dirname(toml_path)
+    project_config = paths.TOMLParser().parse(toml_path, env={"l10n_base": ""})
+    basedir = os.path.join(basedir, project_config.root)
+
+    l10n_files = []
+    if search_path:
+        reldir = search_path
+    else:
+        reldir = basedir
+    for locale in project_config.all_locales:
+        log.info(f"Creating list of files for locale: {locale}.")
+        files = paths.ProjectFiles(locale, [project_config])
+
+        for l10n_file, reference_file, _, _ in files:
+            # Ignore missing files for locale
+            if not os.path.exists(l10n_file):
+                continue
+            # Ignore if reference file does not exist
+            if not os.path.exists(reference_file):
+                continue
+
+            l10n_files.append(
+                {
+                    "abs_path": l10n_file,
+                    "rel_path": os.path.relpath(l10n_file, reldir),
+                }
+            )
+
+    return l10n_files
+
+
+# copy_android_l10n_files {{{1
+def copy_android_l10n_files(l10n_files, src_repo_path, dest_repo_path):
+    """Copy localized files in code repository"""
+
+    log.info(f"Files to copy: {len(l10n_files)}.")
+    for l10n_file in l10n_files:
+        if src_repo_path:
+            src_file = os.path.join(src_repo_path, l10n_file["rel_path"])
+        else:
+            src_file = l10n_file["abs_path"]
+        dest_file = os.path.join(dest_repo_path, l10n_file["rel_path"])
+        log.info(f"  {src_file} -> {dest_file}")
+        # Make sure that the folder exists, then copy file as is
+        os.makedirs(os.path.dirname(dest_file), exist_ok=True)
+        shutil.copy2(src_file, dest_file)
+
+
+# build_commit_message {{{1
+def build_commit_message(description, dontbuild=False, ignore_closed_tree=False):
+    """Build a commit message for android-l10n import.
+
+    Args:
+        dontbuild (bool, optional): whether to add ``DONTBUILD`` to the
+            comment. Defaults to ``False``
+        ignore_closed_tree (bool, optional): whether to add ``CLOSED TREE``
+            to the comment. Defaults to ``False``.
+
+    Returns:
+        str: the commit message
+
+    """
+    approval_str = "r=release a=android_l10n-import"
```

Comment:
Maybe just `a=l10n` like we're doing for desktop bumps.

Reply:
Good idea. Thanks.